nixosModules/hermes/default.nix

```diff
@@ -0,0 +1,44 @@
+{
+  nix-std ? import (fetchTarball "https://github.com/chessai/nix-std/archive/master.tar.gz"),
+  hermes,
+}: {
+  lib,
+  config,
+  pkgs,
+  ...
+}: let
+  hermes-path = lib.meta.getExe hermes;
+  prev = config.services.hermes;
+  cfg = prev // {chains = builtins.map (pkgs.lib.filterAttrsRecursive (_: v: v != null)) prev.chains;};
+  hermes-toml = pkgs.writeTextFile {
+    text = nix-std.lib.serde.toTOML cfg;
+    name = "config.toml";
+  };
+in
+  with lib; {
+    options.services.hermes =
+      import ./options.nix {inherit lib;}
+      // {
+        enable = mkEnableOption "hermes";
+        package = mkOption {
+          type = types.str;
+          default = hermes-path;
+          description = ''
+            The hermes (ibc-rs) software to run.
+          '';
+        };
```

Comment:
why a path instead of a package? seems worse from a types perspective. is it so you can dynamically change the bin for a quick upgrade?

Reply:
let me try to fix it. problem is resuing `cfg` to generate toml.

really it should be may be 
```
{
 package = 
 enable = 
 config = all options etc
}
```

currently it is `//` merges option to root cfg, and cannot render toml out of package.

@JonathanLorimer may be to refactor 

so instead of
```
options.services.hermes =
      import ./options.nix {inherit lib;}
      // {
```

do
```
options.services.hermes = {
      config  = import ./options.nix {inherit lib;}
      enable = ..
      package = ..
}
```

wdt?

Reply:
will do it